src/domain/accounts/services/src/access_token_service_impl.rs

```diff
@@ -0,0 +1,182 @@
+                base32::decode(base32::Alphabet::Crockford, encoded_token)
+            {
+                let token_id = Uuid::from_bytes(
+                    decoded_token_bytes[0..16]
+                        .try_into()
+                        .expect("Invalid uuid length"),
+                );
+                let token_body: [u8; 16] = decoded_token_bytes[16..32]
+                    .try_into()
+                    .expect("Invalid token_bytes_length");
+                let token_checksum: [u8; 4] = decoded_token_bytes[32..36]
+                    .try_into()
+                    .expect("Invalid checksum length");
+                let calculated_checksum = crc32fast::hash(&token_body);
+                if calculated_checksum.to_be_bytes() != token_checksum {
+                    return Err(EncodeTokenError::InvalidTokenChecksum);
+                }
+
+                return Ok(KamuAccessToken {
+                    id: token_id,
+                    prefix: token_prefix.to_string(),
+                    random_bytes: token_body,
+                    random_bytes_hash: Multihash::from_digest_sha3_256(&token_body),
+                    checksum: calculated_checksum,
+                    base32_token: encoded_token.to_string(),
+                    composed_token: access_token,
+                });
+            }
+        }
+        Err(EncodeTokenError::InvalidTokenFormat)
+    }
+}
+
+///////////////////////////////////////////////////////////////////////////////
+
+#[async_trait::async_trait]
+impl AccessTokenService for AccessTokenServiceImpl {
+    async fn create_access_token(
+        &self,
+        token_name: &str,
+        account_id: &AccountID,
+    ) -> Result<KamuAccessToken, CreateAccessTokenError> {
+        let kamu_access_token = self.generate_access_token_impl();
+
+        self.access_token_repository
+            .create_access_token(&AccessToken {
+                id: kamu_access_token.id,
+                token_name: token_name.to_string(),
+                token_hash: kamu_access_token
+                    .random_bytes_hash
+                    .digest()
+                    .try_into()
+                    .unwrap(),
+                created_at: Utc::now(),
+                revoked_at: None,
+                account_id: account_id.clone(),
+            })
+            .await?;
+
+        Ok(kamu_access_token)
+    }
+
+    async fn find_account_id_by_active_token_id(
+        &self,
+        token_id: &Uuid,
+    ) -> Result<Account, GetAccessTokenError> {
+        self.access_token_repository
+            .find_account_by_active_token_id(token_id)
+            .await
+    }
+
+    fn decode_access_token(
+        &self,
+        access_token_str: &str,
+    ) -> Result<KamuAccessToken, EncodeTokenError> {
+        self.decode_token_impl(access_token_str.to_string())
+    }
+
+    fn generate_access_token(&self) -> KamuAccessToken {
+        self.generate_access_token_impl()
+    }
```

Comment:
If you move the "impl" into `KamuAccessToken::new`, you would not need this trait operation at all.